qiskit/quantum_info/synthesis/qsd.py

```diff
@@ -0,0 +1,196 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Quantum Shannon Decomposition.
+
+Method is described in arXiv:quant-ph/0406176.
+"""
+import scipy
+import numpy as np
+from qiskit.circuit import QuantumCircuit, QuantumRegister
+from qiskit.quantum_info.synthesis import two_qubit_decompose, one_qubit_decompose
+from qiskit.circuit.library.standard_gates import CXGate
+from qiskit.quantum_info.operators.predicates import is_hermitian_matrix
+from qiskit.extensions.quantum_initializer.uc_pauli_rot import UCPauliRotGate, _EPS
+
+
+class QuantumShannonDecomposer:
```

Comment:
I see. This is beyond the scope of this PR, but do you think clifford_decompose should have a class as well?

Reply:
It seems we have somewhat of an inconsistent decomposer interface. In addition, cnotdihedral and ion_decompose use the function interface and xx_decompose uses the object interface. I left the object implementation lightweight here so that both options are possible. It would also be easy to make consistent in a future PR if that seems preferable.

Reply:
I'll just remove it then.